### PR TITLE
Removing react dependency in @uifabric/styling library

### LINKED ISSUE #9660: Remove the dependency on React library for @uifabric/icons
I just want to use uifabric/icons library and I am not a React APP. There is an API named getIconClassName in '@uifabric/styling'. I can use it and I can render the icon successfully. 
But to use it, I have to install React library as well. 
Otherwise, build will fail with:

ERROR in ./node_modules/@uifabric/utilities/lib/styled.js
Module not found: Error: Can't resolve 'react' in 'D:\DataIntegrationApp\Product\node_modules\@uifabric\utilities\lib'
ERROR in ./node_modules/@uifabric/utilities/lib/asAsync.js
Module not found: Error: Can't resolve 'react' in 'D:\DataIntegrationApp\Product\node_modules\@uifabric\utilities\lib'
ERROR in ./node_modules/@uifabric/utilities/lib/DelayedRender.js
Module not found: Error: Can't resolve 'react' in 'D:\DataIntegrationApp\Product\node_modules\@uifabric\utilities\lib'
ERROR in ./node_modules/@uifabric/utilities/lib/BaseComponent.js
Module not found: Error: Can't resolve 'react' in 'D:\DataIntegrationApp\Product\node_modules\@uifabric\utilities\lib'
ERROR in ./node_modules/@uifabric/utilities/lib/customizations/customizable.js
Module not found: Error: Can't resolve 'react' in 'D:\DataIntegrationApp\Product\node_modules\@uifabric\utilities\lib\customizations'
ERROR in ./node_modules/@uifabric/utilities/lib/customizations/CustomizerContext.js
Module not found: Error: Can't resolve 'react' in 'D:\DataIntegrationApp\Product\node_modules\@uifabric\utilities\lib\customizations'
ERROR in ./node_modules/@uifabric/utilities/lib/customizations/Customizer.js
Module not found: Error: Can't resolve 'react' in 'D:\DataIntegrationApp\Product\node_modules\@uifabric\utilities\lib\customizations'

Is it possible to remove the dependency of React? Can we just make it a pure Icon Library?

### PR DESCRIPTION
**Pull request by khmakoto**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #9660
- [x] Include a change request file using `$ npm run change`

#### Description of changes

This PR removes unused dependencies in React and React typings in `@uifabric/styling`.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9844)

### CODE REVIEW

File: change/@uifabric-styling-2019-07-17-15-13-36-iconsReact.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "comment": "Removing react dependency in @uifabric/styling library.",
+  "type": "patch",
```
Comment (Vitalius1) [inline change/@uifabric-styling-2019-07-17-15-13-36-iconsReact.json]: Would removing a dependency of a package count as `patch` only? I would go on the safe side and do a minor, but would let you decide. (@kenotron any thoughts?)

**Comment by khmakoto** [comment]
@Vitalius1 I ran `rush update` but the shrinkwrap was not updated. I wonder if it has to do with the fact that the libraries were not actually being used inside the package.

**Comment by Vitalius1** [comment]
I don’t think shrinkwrap cares about whether the dependency is used or not. It just a representation of the dependency tree.

File: change/@uifabric-styling-2019-07-17-15-13-36-iconsReact.json (hunk 1 above)
Comment (cliffkoh) [reply change/@uifabric-styling-2019-07-17-15-13-36-iconsReact.json]: Removing dependency like this should be fine as patch

**Pull request merged**